frontend/packages/kubevirt-plugin/src/k8s/enhancedK8sMethods/__tests__/enhancedK8sMethods.spec.ts

```diff
@@ -0,0 +1,205 @@
+    .setMemory('1', 'Mi')
+    .asResource();
+
+  const testPatch = new PatchBuilder('/')
+    .add({
+      additionalData: 'testData',
+    })
+    .build();
+
+  it('records history and shows actualState', async () => {
+    const { methods, methodsUnchecked } = spyEnhancedK8sMethods();
+
+    expect(methods.getHistory()).toHaveLength(0);
+    expect(methods.getActualState()).toHaveLength(0);
+
+    await methods.k8sCreate(VirtualMachineModel, testVM);
+    await methods.k8sPatch(VirtualMachineModel, testVM, [testPatch]);
+    await methods.k8sKill(VirtualMachineModel, testVM);
+    const history = methods.getHistory();
+
+    expectHistory(history, [
+      new HistoryItem(HistoryType.CREATE, testVM),
+      new HistoryItem(HistoryType.PATCH, { ...testVM, ...testPatch }),
+      new HistoryItem(HistoryType.DELETE, testVM),
+    ]);
+
+    expect(methodsUnchecked.history).toHaveLength(3);
+  });
+
+  it('shows actualState', async () => {
+    const { methods, methodsUnchecked } = spyEnhancedK8sMethods();
+
+    await methods.k8sCreate(VirtualMachineModel, testVM);
+    await methods.k8sPatch(VirtualMachineModel, testVM, []);
+    await methods.k8sPatch(VirtualMachineModel, testVM, []);
+    await methodsUnchecked.k8sGet(VirtualMachineModel, testVM);
+    await methods.k8sKill(VirtualMachineModel, testVM);
+    await methods.k8sCreate(VirtualMachineModel, otherTestVM);
+    await methods.k8sPatch(VirtualMachineModel, otherTestVM, []);
+    await methodsUnchecked.k8sGet(VirtualMachineModel, otherTestVM);
+    await methods.k8sCreate(VirtualMachineModel, testVM);
+    await methodsUnchecked.k8sGet(VirtualMachineModel, testVM);
+    await methods.k8sPatch(VirtualMachineModel, testVM, [testPatch]);
+    const actualState = methods.getActualState();
+
+    expect(actualState).toHaveLength(2);
+    expect(actualState[0]).toEqual(otherTestVM);
+    expect(actualState[1]).toEqual({ ...testVM, ...testPatch });
+  });
+
+  it('disables recording history', async () => {
+    const { methods, methodsUnchecked } = spyEnhancedK8sMethods();
+
+    await methods.k8sCreate(VirtualMachineModel, testVM, null, disableHistoryOpts);
+    await methods.k8sPatch(VirtualMachineModel, testVM, [testPatch], disableHistoryOpts);
+    await methods.k8sKill(VirtualMachineModel, testVM, null, null, disableHistoryOpts);
+
+    expect(methods.getHistory()).toHaveLength(0);
+    expect(methodsUnchecked.history).toHaveLength(0);
+
+    await methods.k8sCreate(VirtualMachineModel, testVM);
+    expect(methods.getHistory()).toHaveLength(1);
+  });
+
+  it('rollback', async () => {
+    const { methods } = spyEnhancedK8sMethods();
+
+    await methods.k8sCreate(VirtualMachineModel, testVM);
+    await methods.k8sPatch(VirtualMachineModel, testVM, []);
+    await methods.k8sCreate(VirtualMachineModel, otherTestVM);
+
+    const results = await methods.rollback();
+
+    expect(results).toHaveLength(2);
+    expect(methods.getActualState()).toHaveLength(0);
```

Comment:
why is `getActualState()` empty? just to understand..

Reply:
because the rollback worked and there are no living objects we created before on the cluster